src/event-log.c

```diff
@@ -0,0 +1,405 @@
+			value = ifield->value;
+		} else if (g_strcmp0(ifield->key, "BUNDLE_VERSION") == 0) {
+			key = "bundle version";
+			value = ifield->value;
+		} else if (g_strcmp0(ifield->key, "RAUC_SLOT") == 0) {
+			key = "slot";
+			value = ifield->value;
+		} else if (g_strcmp0(ifield->key, "SLOT_BOOTNAME") == 0) {
+			key = "bootname";
+			value = ifield->value;
+		}
+
+		if (key) {
+			g_string_append_printf(gstring, "\n                      %s: %s", key, value);
+		}
+	}
+
+	return g_string_free(gstring, FALSE);
+}
+
+/**
+ * Rotates log files.
+ *
+ * If a maximum size is configured and the loggers current size + provided
+ * 'next_len' exceed this maximum size, the method will close the current
+ * output stream, and rename the file to add a .1 suffix.
+ *
+ * All previously existing rotation files will be rotated by +1.
+ *
+ * If max-files is set, rotation is limited to max-files files.
+ *
+ * @param logger Logger to rotate file for
+ * @param next_len Length of next data to add.
+ *        This is used to determine if the log file needs to be rotated.
+ * @param error return location for a GError or NULL.
+ *
+ * @return TRUE if rotation succeeded (or was a no-op). FALSE in case of a failure.
+ */
+static gboolean rotate_logfiles(REventLogger* logger, goffset next_len, GError **error)
+{
+	g_autoptr(GFile) logfile = NULL;
+	g_autofree gchar *rotfile = NULL;
+	g_autoptr(GError) ierror = NULL;
+
+	g_return_val_if_fail(logger, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	/* method is no-op if no maxsize is configured */
+	if (!logger->maxsize)
+		return TRUE;
+
+	/* return if size limit not reached yet */
+	if (logger->filesize + next_len <= logger->maxsize)
+		return TRUE;
+
+	/* Close open stream */
+	g_clear_pointer(&logger->logstream, g_object_unref);
+
+	/* iterate through the list of potentially existing rotation files and move
+	 * all (existing ones) but the last one. This will override (and thus drop)
+	 * the last rotation file */
+	g_assert(logger->maxfiles >= 1);
+	for (guint file_num = logger->maxfiles - 1; file_num > 0; file_num--) {
+		g_autofree gchar *from_file = g_strdup_printf("%s.%d", logger->filename, file_num - 1);
+		g_autofree gchar *to_file = g_strdup_printf("%s.%d", logger->filename, file_num);
+
+		/* rotate .N-1 to .N */
+		if (g_rename(from_file, to_file) == -1) {
+			int err = errno;
+			if (err == ENOENT) {
+				/* skip non-existing 'from' files */
+				continue;
+			} else {
+				g_set_error(error, G_FILE_ERROR, g_file_error_from_errno(err), "Failed to rotate log file: %s", g_strerror(err));
+				return FALSE;
+			}
+		}
+	}
+
+	/* rotate current log file as next .1 file */
+	rotfile = g_strdup_printf("%s.1", logger->filename);
+	if (g_rename(logger->filename, rotfile) == -1) {
+		int err = errno;
+		g_set_error(error, G_FILE_ERROR, g_file_error_from_errno(err), "Failed to rotate log file: %s", g_strerror(err));
+		return FALSE;
+	} else {
+		logger->filesize = 0;
+	}
```

Comment:
It doesn't really make sense to have an `else` here after the error return above.

The `filesize` reinit belongs in the re-open part below.

Reply:
Yes, absolutely makes sense.